apps/documents/serializers.py

```diff
@@ -80,11 +84,22 @@ def get_presigned_s3(self, obj):
         if obj.upload_status != UploadStatus.COMPLETE:
             return super().get_presigned_s3(obj)
 
+        key = '/'.join(obj.s3_path.split('/')[3:])
+
+        try:
+            settings.S3_CLIENT.head_object(Bucket=settings.S3_BUCKET, Key=key)
+        except ClientError:
+            # The document doesn't exist anymore in the bucket. The bucket is going to be repopulated from vault
+            storage_credentials_id, wallet_id, vault_id, filename = key.split('/', 3)
```

Comment:
Instead of doing this you should be able to just directly use the properties that are referenced in the s3_key property in the Document model